src/main/java/roboy/context/Context.java

```diff
@@ -1,135 +1,197 @@
 package roboy.context;
 
-import com.google.common.collect.ImmutableClassToInstanceMap;
 import roboy.context.contextObjects.*;
+import roboy.memory.nodes.Interlocutor;
+import roboy.ros.RosMainNode;
+import roboy_communication_cognition.DirectionVector;
 
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 
 /**
  * Singleton class serving as an interface to access all context objects.
- * Takes care of correct initialization of attribute histories and updaters.
- * <p>
- * Queries to values are handled through the inherited AttributeManager methods.
- * <p>
+ * Takes care of correct initialization.
  * For usage examples, check out ContextTest.java
  */
-public class Context extends AttributeManager<Context.ValueHistories, Context.Values> {
-    private static Context context;
+public class Context {
+    private static final Object initializationLock = new Object();
 
-    private final ArrayList<ExternalUpdater> externalUpdaters = new ArrayList<>();
-    public final HashMap<Class, InternalUpdater> internalUpdaters = new HashMap<>();
+    /* VALUES INITIALIZED HERE */
+    public final ValueInterface<FaceCoordinates, CoordinateSet> FACE_COORDINATES =
```

Comment:
I agree that this is verbose, and at least somewhat redundant (the third type prevents type casting within tests).

The idea behind these empty classes is to define specific context objects, especially their type parameters, somewhere outside of the Context main class. I believe it has advantages in clarity, i.e. it is easier to understand what's going on behind the scenes in Context. Another advantage is that the class names are used by the Context GUI, but this is not too tragic.

However, creating empty classes is not mandatory. Take dialog topics as example:

    public final HistoryInterface<DialogTopics, Integer, String> DIALOG_TOPICS =
            new HistoryInterface<>(new DialogTopics());

It is possible to write them like this:

    public final HistoryInterface<ValueHistory<String>, Integer, String> DIALOG_TOPICS =
        new HistoryInterface<>(new ValueHistory<>());

And with your suggested changes it could look like this:

    public final HistoryInterface<Integer, String> DIALOG_TOPICS =
        new HistoryInterface<>(new ValueHistory<>());

Using general classes, we omit information about the History itself, and might confuse anyone who is new to the Context architecture. One would also need to be more careful defining the types. But it does look short and neat, I have to admit that ;)

In conclusion - technically, I could remove the third parameter. This would require modifying the Context class and some tests, but change nothing about the current usage of Context by other classes. I'd prefer not making the changes in the scope of this PR before further discussion, though.

Reply:
It is very not ideal that we now have this open question about context but are still merging to master. We gotta resolve it asap and before the new students touch the code, i.e. Mid next week